identity/src/main/java/com/stripe/android/identity/analytics/IdentityAnalyticsRequestFactory.kt

```diff
@@ -25,34 +28,77 @@ internal class IdentityAnalyticsRequestFactory @Inject constructor(
+        }
+
+        return mapOf(
             PARAM_VERIFICATION_SESSION to args.verificationSessionId,
-            PARAM_EVENT_META_DATA to
-                mutableMapOf(
-                    *pairs
-                ).also {
-                    verificationPage?.let {
-                        PARAM_LIVE_MODE to it.livemode
-                    }
+            PARAM_EVENT_META_DATA to metadataMap
+        )
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private suspend fun maybeLogExperimentAndSendLog(
+        eventName: String,
+        additionalParams: Map<String, Any> = mapOf()
+    ) {
+        verificationPage?.let { verificationPage ->
+            val experiments = verificationPage.experiments
+            val userSessionId = verificationPage.userSessionId
+            val metaDatas = additionalParams.getOrDefault(PARAM_EVENT_META_DATA, null) as Map<String, Any>?
+
+            for (exp: VerificationPageStaticContentExperiment in experiments) {
+                if (exp.matches(eventName, metaDatas)) {
+                    identityRepository.sendAnalyticsRequest(
+                        requestFactory.createRequest(
+                            eventName = EVENT_EXPERIMENT_EXPOSURE,
+                            additionalParams = mapOf(
+                                PARAM_ARB_ID to userSessionId,
+                                PARAM_EXPERIMENT_RETRIEVED to exp.experimentName
+                            )
+                        )
+                    )
                 }
+            }
+        }
+        val request = requestFactory.createRequest(
+            eventName = eventName,
+            additionalParams = additionalParams
         )
+        identityRepository.sendAnalyticsRequest(request)
+    }
+
+    private fun VerificationPageStaticContentExperiment.matches(
+        eventName: String,
+        metaDatas: Map<String, Any>?
+    ): Boolean {
+        return if (this.eventMetadata.isEmpty()) {
+            this.eventName == eventName && metaDatas == null
+        } else {
+            metaDatas?.let {
+                this.eventName == eventName && metaDatas.entries.containsAll(this.eventMetadata.entries)
+            } ?: false
+        }
+    }
 
-    fun sheetPresented() = requestFactory.createRequest(
-        EVENT_SHEET_PRESENTED,
-        mapOf(
+    suspend fun sheetPresented() = maybeLogExperimentAndSendLog(
+        eventName = EVENT_SHEET_PRESENTED,
+        additionalParams = mapOf(
             PARAM_VERIFICATION_SESSION to args.verificationSessionId
         )
     )
 
-    fun sheetClosed(sessionResult: String) = requestFactory.createRequest(
+    suspend fun sheetClosed(sessionResult: String) = maybeLogExperimentAndSendLog(
         eventName = EVENT_SHEET_CLOSED,
         additionalParams = additionalParamWithEventMetadata(
             PARAM_SESSION_RESULT to sessionResult
         )
     )
 
-    fun verificationSucceeded(
+    suspend fun verificationSucceeded(
```

Comment:
You can also avoid making all these suspended by using GlobalScope in `maybeLogExperimentAndSendLog` (making sure it's error safe to prevent crashes)